x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/string/StringFunctionUtils.java

```diff
@@ -47,11 +48,16 @@ static String trimTrailingWhitespaces(String s) {
             return s;
         }
 
-        StringBuilder sb = new StringBuilder(s);
-        while (sb.length() > 0 && Character.isWhitespace(sb.charAt(sb.length() - 1))) {
-            sb.deleteCharAt(sb.length() - 1);
+        int endIdx = -1;
+        for (int i = s.length() - 1; (i >= 0) && (endIdx < 0); i--) {
+            if (Character.isWhitespace(s.charAt(i)) == false) {
+                endIdx = i;
+            }
         }
-        return sb.toString();
+        if (endIdx < 0) {
+            return EMPTY;
+        }
+        return s.substring(0, endIdx + 1);
```

Comment:
Nit: more compact alternative.

```
int i = s.length() - 1;
while (i >= 0 && Character.isWhitespace(s.charAt(i))) {
    i --;
}
return s.substring(0, i + 1);
```

Similar for treamLeading.

Reply:
@bpintea cool, thx, with your approach:
```
Benchmark                         Mode  Cnt      Score    Error  Units
StringTrim.testWithJavaStrip      avgt   25   1342.251 ±  7.019  ns/op
StringTrim.testWithStringBuilder  avgt   25  82599.137 ± 86.073  ns/op
StringTrim.testWithSubstring      avgt   25   1035.607 ± 15.208  ns/op
```